File: API_DBFruits/API_DBFruits/Models/GioHang.cs
using System;
using System.Collections.Generic;

namespace API_DBFruits.Models;

public partial class GioHang
{
    public int GioHangId { get; set; }

    public String? KhachHangId { get; set; }

    public int? SanPhamId { get; set; }
    public int? SoLuong { get; set; }

    public DateOnly ThoiGian { get; set; }

    public virtual User? KhachHang { get; set; }

    public virtual SanPham? SanPham { get; set; }
}

File: lib/models/GioHang.dart
class GioHang {
  GioHang({
    required this.gioHangId,
    required this.khachHangId,
    required this.sanPhamId,
    required this.soLuong,
    required this.thoiGian,
  });

  final int? gioHangId;
  final String? khachHangId;
  final int? sanPhamId;
  final int? soLuong;
  final DateTime? thoiGian;

  factory GioHang.fromJson(Map<String, dynamic> json) {
    return GioHang(
      gioHangId: json["gioHangId"],
      khachHangId: json["khachHangId"],
      sanPhamId: json["sanPhamId"],
      soLuong: json["soLuong"],
      thoiGian: DateTime.tryParse(json["thoiGian"] ?? ""),
    );
  }
}
